pkg/controller/disconnected_operator.go

```diff
@@ -0,0 +1,137 @@
+package controller
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"time"
+
+	"k8s.io/klog/v2"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/pkg/version"
+	"k8s.io/client-go/rest"
+
+	configv1client "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
+	"github.com/openshift/library-go/pkg/controller/controllercmd"
+
+	"github.com/openshift/insights-operator/pkg/config"
+	"github.com/openshift/insights-operator/pkg/config/configobserver"
+	"github.com/openshift/insights-operator/pkg/controller/periodic"
+	"github.com/openshift/insights-operator/pkg/controller/status"
+	"github.com/openshift/insights-operator/pkg/gather"
+	"github.com/openshift/insights-operator/pkg/gather/clusterconfig"
+	"github.com/openshift/insights-operator/pkg/recorder"
+	"github.com/openshift/insights-operator/pkg/recorder/diskrecorder"
+)
+
+type Disconnected struct {
+	config.Controller
+}
+
+// LoadConfig unmarshalls config from obj and loads it to this Support struct
+func (d *Disconnected) LoadConfig(obj map[string]interface{}) error {
+	var cfg config.Serialized
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(obj, &cfg); err != nil {
+		return fmt.Errorf("unable to load config: %v", err)
+	}
+
+	controller, err := cfg.ToDisconnectedController(&d.Controller)
+	if err != nil {
+		return err
+	}
+	d.Controller = *controller
+
+	data, _ := json.Marshal(cfg)
```

Comment:
Are you sure the error will never happen?

Reply:
good question, for now its mainly a copy paste from the `operator.go` in the most part. Will look into it

Reply:
It shouldn't reach this point if something was wrong with `cfg` and also its only for printing it to the log. :)